report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,748 @@
+HCL allows for the use of ';' and '{\}n' for line endings.
+for simplicity, will the abstract syntax and rules in this section utilize ';' for line endings. 
+
+\subsection{Scoping of HCL}
+The scope rules of a language determines which variable environment and procedure environment are being accessed, when using the name of a variable or function name.
+Since the semantics of HCL will be stated using operational structural semantics, the scope rule can be shown by simple accessing the environment required by the scope rules.
+The scope rules of HCL are fully static.
+This means that the bindings accessed when using variable or functions names, will be the ones that were bound when the variable or function was declared.
+The reader should note that, since HCL is utilizing fully static scope rules, transitions for function calls needs special components to successfully enable recursive calling of function.
+In other words, when make a recursive call in HCL the bindings of the procedure environment will be dynamic.
+
+\subsection{Abstract Syntax}
+The abstract syntax will now be presented.
+For convenience, the following notational conventions will be followed throughout this section.
+
+In the rest of the section $t1, .
+.
+.
+ , tn$ will be abbreviated with $t\textasciitilde$ and $t1 .
+ .
+ .
+  tn$ with $t^*$ ($t^+$ in the non-empty case).
+
+The syntax categories and meta-variables used in the semantic can be seen in table \ref{tab:metaVarab}. 
+\begin{center}
+	\begin{table}[ht]
+		\centering
+		\caption{The syntax categories and meta-variables used in the operational semantics of HCL}
+		\label{tab:metaVar}
+		\begin{tabular}{ll}
+			\textbf{Metavariable} & \textbf{Name} \\
+			$a \in A_{exp}$ 	  & Arithmetic expression \\
+			$n \in Num$			  & Number Literal \\
+			$x \in Var$           & Variable \\
+			$b \in B_{exp}$		  & Boolean expression\\ 
+			$tf \in {True,False}$ & Boolean literal \\
+			$t \in Txt_{exp}$     & Text expression\\
+			$str \in {\Sigma^*}$  & String Literal \\
+			$list \in list_{exp}$ & List expression \\
+			$lst \in List$		  & List literal\\
+			$tuple \in tpl_{exp}$ & Tuple expression\\
+			$tpl \in Tuple$	      & Tuple Literal\\
+			$L \in Lambda$		  & Lambda expression\\
+			$e \in Exp$			  & Expression\\
+			$el \in Lit$		  & Expression Literal\\
+			$p \in Pnames$        & Procedure names\\
+			$L \in Lambda$        & Statement Lambdas\\
+			$S \in Stm$           & Statement\\
+			$f \in Fnames$ 		  & Function names\\
+			$D_V \in Dcl_V$		  & Variable declarations\\
+			$D_P \in Dcl_P$       & Function declarations\\
+			$T \in Type$          & Types
+		\end{tabular}
+	\end{table}
+\end{center}
+
+For the rest of this section, it will be assumed that functions exists with domains equal to the sets of literals associated with the various expressions' syntax categories, and with co-domains equal to the values extracted from those literals.
```

Comment:
make sure this is achored below the above table.

Reply:
Fixed